Composer/packages/client/src/recoilModel/Recognizers.tsx

```diff
@@ -148,40 +165,43 @@ export const Recognizer = React.memo((props: { projectId: string }) => {
   const qnaFiles = useRecoilValue(qnaFilesState(projectId));
   const settings = useRecoilValue(settingsState(projectId));
   const curRecognizers = useRecoilValue(recognizersSelectorFamily(projectId));
+  const curRecognizersRef = useRef(curRecognizers);
   const filePersistence = useRecoilValue(filePersistenceState(projectId));
+  curRecognizersRef.current = curRecognizers;
```

Comment:
Not sure if we should move this to the `useEffect` just to be safe?

Reply:
I'm afraid that the setRecognizer will trigger the  calculation again if I put it to the useEffect dependency